Documentation/sourcebuild-in-repos/new-repo.md

```diff
@@ -302,26 +304,53 @@ will detect and publish.
 
-## Add Downstream Dependency
+## Incorporate the new repo into the Source-Build dependency tree
 
-The graph defined by `eng/Version.Details.xml` defines what is included
-in source build. This dependency graph starts at
+Once your repo can be source-built, it is time to register it into the
+Source-Build dependency tree. The graph of the product is defined by the
+`eng/Version.Details.xml` files. This dependency graph starts at
 [dotnet/installer](https://github.com/dotnet/installer/blob/main/eng/Version.Details.xml).
-The .NET source build insfrastructure walks this graph and determines
-the full set of repos to build. Source build will only build one copy/commit
-of each repo (the first commit found in the dependency graph). Therefore,
-when adding a new repositories, a dependency must be created within the
-`eng/Version.Details.xml` graph. To do this, go to the repo which depends on
-the new repo and add a
-[new source build dependency](#engversiondetailsxml) to the new source build
-repo.
+The dependendecies of repos declared in these files are walked and the
+first copy/commit of each repo found in the dependency graph is used.
+
+Therefore, when adding new repositories, a dependency must be created within
+the `eng/Version.Details.xml` graph. To do this, **go to the repo which depends
+on the new repo and add a [new source build dependency](#engversiondetailsxml)
+to the new source build repo**.
+
+## Source build repos and the VMR
+
+Another effect of adding a new source build repository is that its sources
+will be synchronized into the [Virtual Monolithic Repository of .NET](https://github.com/dotnet/dotnet).
+The VMR is then where the official Source-Build happens from. The sources
+are synchronized once the associated commit/package flows into `dotnet/installer`.
+
+In order for the sources of the new repo to by synchronized into the VMR,
+the repo needs to be registered in the
+[`source-mappings.json` file](https://github.com/dotnet/dotnet/blob/main/src/source-mappings.json)
+which tells the tooling where from and which sources should be synchronized.
+Please open a PR in [`dotnet/dotnet`](https://github.com/dotnet/dotnet) and
+add your repository. The name must match the name declared in the `SourceBuild`
+tag in `Version.Details.xml` created in the previous step.
+
+### Cloaking (filtering) the repository sources
+
+When creating the `source-mappings.json` record for the new repo, there is
+a possibility of filtering which sources get synchronized into the VMR.
+The VMR should only really contain plain text-based files as it is consumed
+by 3rd party .NET distro builders who do not accept any non-text sources
+(e.g. binaries). When registering the repository into the VMR, it is a good
+time to consider which files are required for it to build and only synchronize
+those. Commonly, repositories contain binaries that are required for testing
+or similar purposes. Files like these should not be synchronized into the VMR.
```

Comment:
FWIW, I think it's absolutely okay if there are specially crafted binaries that are used for testing - eg, a hand crafted dll that might contain invalid bytecode that verifies that the runtime can handle it correctly. On the other hand, including a binary for an external tool (eg, gradle) is probably a bad idea

Reply:
Ah, my understanding is that it would be super difficult to tell how a binary is used so no binaries are allowed at all.

We can discuss this at the next meeting then?